docs: add docs for the build-docs command (#871)
Redocly/redocly-cli: docs/commands/build-docs.md

```

## Options

Option | Type | Description
-- | -- | --
api | string | Path to the API definition filename or alias that you want to generate the build for. Refer to [the api section](#api) for more options.
--output | string | Output file. Default value is `redoc-static.html`.
--title | string | Page title.
--disableGoogleFont | boolean | Disable Google font. Default value is `false`.
--cdn | boolean | Use latest Redoc version. Default value is `false`.
--template | string | Uses custom [Handlebars](https://handlebarsjs.com/) templates to render your OpenAPI definition.
--templateOptions | string | Adds template options you want to pass to your custom Handlebars template. To add options, use dot notation.
--features.openapi | string | Customizes your output using [Redoc functionality options](https://redocly.com/docs/api-reference-docs/configuration/functionality) or [Redoc theming options](https://redocly.com/docs/api-reference-docs/configuration/theming).
--config | string | Specify path to the [configuration file](#custom-configuration-file).
--help | boolean | Show help.
--version | boolean | Show version number.

## Examples

> ### Api

Reviewer: ```suggestion
### API argument
```

# `build-docs`

## Introduction

The `build-docs` command builds Redoc into a zero-dependency HTML file.

## Usage

```bash
redocly build-docs <api>
redocly build-docs <api> --output=custom.html
redocly build-docs <api> --features.openapi.theme.colors.primary.main=orange

> redocly build-docs <api> -t custom.hbs

Reviewer: `-t` isn't documented anywhere? Are there other option shortcuts?

Reviewer: ```suggestion
redocly build-docs <api> --template custom.hbs
```
Maybe I would recommend this for now.

Author: I forgot to add the aliases to the documentation.
Added. We can use both cases.
I am not sure about the best approach.

redocly build-docs <api> -t custom.hbs --templateOptions.metaDescription "Page meta description"
```

## Options

Option | Type | Description
-- | -- | --
api | string | Path to the API definition filename or alias that you want to generate the build for. Refer to the [API examples](#api-examples) for more information.
--output | string | Sets the path and name of the output file. The default value is `redoc-static.html`.
--title | string | Sets the page title.
--disableGoogleFont | boolean | Disables Google fonts. The default value is `false`.
--cdn | boolean | Uses the CDN pointing to the latest version of Redoc. If not enabled, the Redoc version is selected from Redocly CLI's associated dependency (and to update Redoc requires updating the CLI and building the docs again). The default value is `false`.
--template | string | Uses custom [Handlebars](https://handlebarsjs.com/) templates to render your OpenAPI definition.
--templateOptions | string | Adds template options you want to pass to your custom Handlebars template. To add options, use dot notation.

> --features.openapi | string | Customizes your output using [Redoc functionality options](https://redocly.com/docs/api-reference-docs/configuration/functionality) or [Redoc theming options](https://redocly.com/docs/api-reference-docs/configuration/theming).

Reviewer: ⚠️ Inconsistency alert: the other options use camelCase. This one uses a dot-separator. This one is consistent with the naming in the configuration file. Should anything change? Should the configuration file change to `featuresOpenapi`? Thoughts? cc @RomanHotsiy

Reviewer 2: hmmm... Good question... Maybe `openapiFeatures`?

Reviewer 3: Shouldn't it be consistent with `config.yaml` file? 
That was previously discussed [here](https://github.com/Redocly/redocly-cli/pull/871#issuecomment-1257571473)

Reviewer: We talked and suggested `oasFeatures`, as well as updating the config file format to match. And eventually, `graphqlFeatures`, etc...

Author: added for `build-docs` command in [PR](https://github.com/Redocly/redocly-cli/pull/892)

Reviewer: @RomanHotsiy and I talked more and maybe it will be `features`?

Reviewer: And in this case, it may be a nested object, and therefore features.openapi.hideTryIt would map to:

```
features:
  openapi:
     hideTryIt: true
```

Reviewer: We looked at this in more depth, and believe it should be:

```yaml
theme:
  openapi:
    hideTryIt: true
```

# `build-docs`

## Introduction

The `build-docs` build api and Redoc into a zero-dependency HTML file.

## Usage

```bash
redocly build-docs <api>
redocly build-docs <api> --output=custom.html
redocly build-docs <api> --options.theme.colors.primary.main=orange
redocly build-docs <api> -t custom.hbs

> redocly build-doc <api> -t custom.hbs --templateOptions.metaDescription "Page meta description"

Reviewer: ```suggestion
redocly build-docs <api> -t custom.hbs --templateOptions.metaDescription "Page meta description"
```

```

## Options

Option | Type | Description
-- | -- | --

> api | string | **REQUIRED.** path or URL to your to the API definition file.

Reviewer: Does it work to pass in the name from the configuration file:

```
apis:
  apiname@v1:
    root: ...
```

`redocly build-docs apiname@v1`?

Author: no, it doesn't work for now. We should improve that later. 
Firstly we want to deprecate `redoc-cli`.
In the first version of this PR, it contains `<spec>`.
I talked with @RomanHotsiy , and he recommended renaming `<spec>` to `<apis>` for docs consistency.
What do you think about that? Should we rename that to `<spec>`?

Reviewer: I think this introduces inconsistency into commands actually. Please hold off on doing this. 

All commands should accept `apis` which is a list of file paths or aliases from the configuration file.

> api | string | Path to the API definition filename or alias that you want to generate the preview for. Refer to [the api section](#api) for more options.

Reviewer: Is this required?

Reviewer: What is "the api section" and there is none that I found.

Author: > Is this required?

No. We support running `build-docs` command in the project directory if this directory contains a config file. We try to find the config file in the working directory.
```
redocly build-docs
```

> What is "the api section" and there is none that I found.

Added API section to documentation, like we have in `preview-docs` command.

--output | string | Output file. Default value is `redoc-static.html`.

> --title | string | Page Title.

Reviewer: ```suggestion
--title | string | Page title.
```

--disableGoogleFont | boolean | Disable Google Font. Default value is `false`.

> --cdn | boolean | Use latest ReDoc version. Default value is `false`.

Reviewer: ```suggestion
--cdn | boolean | Use latest Redoc version. Default value is `false`.
```

> --disableGoogleFont | boolean | Disable Google Font. Default value is `false`.

Reviewer: ```suggestion
--disableGoogleFont | boolean | Disable Google font. Default value is `false`.
```

> --cdn | boolean | Use latest Redoc version. Default value is `false`.

Reviewer: I'm not sure I understand this. What version is used otherwise?

Author: If we run the `build-docs` command without `cdn` option we use the Redoc version from `package.json` inside `@redocly/cli` package.
Respectively, if we pass `cdn` option we'll use the latest version of Redoc.

This option we have in `redoc-cli` and migrated here for backward compatibility. Many users used this option in `redoc-cli`.

Reviewer: ```suggestion
--cdn | boolean | Uses the CDN pointing to the latest version of Redoc. If not enabled, the Redoc version is selected from Redocly CLI's associated dependency (and to update Redoc requires updating the CLI and building the docs again). Default value is `false`.
```

--template | string | Uses custom [Handlebars](https://handlebarsjs.com/) templates to render your OpenAPI definition.

> --templateOptions | string | Adds template options you want to pass to your custom Handlebars template. To add options, use dot notation.

Reviewer: Can you provide an example? I don't understand how this connects to dot notation and handlebars template options.

Author: The command is  (we have this in `Usage` section)
```bash
redocly build-docs <api> -t custom.hbs --templateOptions.metaDescription "Page meta description"
```
 And below we have `Sample Handlebars template` which contains line 
 ```
 <meta description="{{{templateOptions.metaDescription}}}">
 ```
 
That means options `templateOptions.metaDescription` from command we passed to HTML template.

--options | string | Customizes your output using [Redoc functionality options](https://redocly.com/docs/api-reference-docs/configuration/functionality) or [Redoc theming options](https://redocly.com/docs/api-reference-docs/configuration/theming).
--help | boolean | Show help.
--version | boolean | Show version number.

### Examples

#### Build docs

Build docs with the main color changed to `orange`:

```bash

> redocly build-doc openapi.yaml --options.theme.colors.primary.main=orange

Reviewer: ```suggestion
redocly build-docs openapi.yaml --options.theme.colors.primary.main=orange
```

```

Build docs using a custom Handlebars template and add custom `templateOptions`:

```bash

> redocly build-doc http://petstore.swagger.io/v2/swagger.json -t custom.hbs --templateOptions.metaDescription "Page meta description"

Reviewer: ```suggestion
redocly build-docs http://petstore.swagger.io/v2/swagger.json -t custom.hbs --templateOptions.metaDescription "Page meta description"
```
